lib/dry/web/parser.rb

```diff
@@ -0,0 +1,54 @@
+module Dry
+  module Web
+    class Parser
+      # Regex extracted from dotenv gem
+      # https://github.com/bkeepers/dotenv/blob/master/lib/dotenv/parser.rb#L14
+      LINE = %r(
+        \A
+        \s*
+        (?:export\s+)?    # optional export
+        ([\w\.]+)         # key
+        (?:\s*=\s*|:\s+?) # separator
+        (                 # optional value begin
+          '(?:\'|[^'])*'  #   single quoted value
+          |               #   or
+          "(?:\"|[^"])*"  #   double quoted value
+          |               #   or
+          [^#\n]+         #   unquoted value
+        )?                # value end
+        \s*
+        (?:\#.*)?         # optional comment
+        \z
+      )x
+
+      def self.call(file)
+        new(file).parse
+      end
+
+      def initialize(file)
+        @file = file
+        @hash = {}
+      end
+
+      def parse
+        File.readlines(@file).each do |line|
+          parse_line(line)
+        end
+        @hash
+      end
```

Comment:
Just regarding the design of this object, unless there's a good reason to keep local state, I feel like it's a good approach if we make these stateless.

So for here it'd mean something like this:

```ruby
parser = Parser.new # new parser, ready to operate on any file
parser.(file) # returns a hash of the keys/values parsed from the file
```

I think this'd probably make the rest of the code within this class slightly cleaner as well (not having ivars like `@hash` that we refer to or manipulate from various different places.

Let me know if that's enough for you to go on, or if you'd like more concrete suggestions here :)